spec/lib/appsignal/transaction_spec.rb

```diff
@@ -200,6 +208,61 @@
         end
       end
     end
+
+    context "when a transaction has additional errors" do
+      let(:error) do
+        e = ExampleStandardError.new("test message")
+        allow(e).to receive(:backtrace).and_return(["line 1"])
+        e
+      end
+
+      let(:other_error) do
+        e = ExampleStandardError.new("other test message")
+        allow(e).to receive(:backtrace).and_return(["line 2"])
+        e
+      end
+
+      context "when an error is already set on the transaction" do
+        it "reports additional errors as duplicate transactions" do
+          transaction.set_error(error)
+          transaction.add_error(other_error)
+
+          expect(transaction).to receive(:duplicate).and_wrap_original do |original_method|
+            duplicate_transaction = original_method.call
+            expect(duplicate_transaction).to receive(:set_error).with(other_error)
+            expect(duplicate_transaction).to receive(:complete)
+            duplicate_transaction
+          end
```

Comment:
I can help with writing some helpers for this. I do not want us to start asserting method calls again after I just refactored them all out.

Reply:
Feel free to give it a try! I'm not sure how we could write helpers for this that check against the transaction's `.to_h` -- this all happens when calling `transaction.complete`, meaning that the extension will return an empty hash after this is done.

Reply:
Turns out it already works with the existing helpers so I don't need to make new helpers.

Reply:
I made a PR here: https://github.com/appsignal/appsignal-ruby/pull/1212